### ISSUE TITLE
No .on() - listener method when using typescript

### ISSUE THREAD

**Issue by ishiharas** [JiPaix/xdccJS#493]
Hello, yesterday I tried to implement a couple of things with your library.
I decided to go by the typescript flow.

The problem is, neither the XDCC-class, nor the Job-class expose listener methods.

<img width="442" alt="Bildschirmfoto 2022-06-04 um 15 38 29" src="https://user-images.githubusercontent.com/43359029/172003003-312d988c-5082-4243-abba-469cf35d7185.png">

<img width="286" alt="Bildschirmfoto 2022-06-04 um 15 40 39" src="https://user-images.githubusercontent.com/43359029/172003249-499991a1-23ae-4e84-a4af-a69976495c6e.png">

The only way right now to listen for "download" or "downloading" events is to go through the .irc object.

<img width="733" alt="Bildschirmfoto 2022-06-04 um 15 42 14" src="https://user-images.githubusercontent.com/43359029/172003295-70a26c9c-22f8-4647-824a-215ed3b8604d.png">

**Comment by JiPaix** [comment]
autocomplete's working using xdccJS `4.4.19` (latest) + typescript `4.6.3` (vscode).

<details><summary>change typescript version in vscode</summary>

`CTRL+SHIFT+P`
![image](https://user-images.githubusercontent.com/26584973/172003852-728258b3-7f69-48b9-b893-a381f2214e38.png)
![image](https://user-images.githubusercontent.com/26584973/172003911-5edfac96-021a-4431-948b-99d293eb062e.png)

</details>

<details><summary>xdccJS listeners</summary>

![screen1](https://user-images.githubusercontent.com/26584973/172003626-2dfdc056-abcc-46e6-b30f-b702db90e822.png)

</details>
<details><summary>Job listeners</summary>

![image](https://user-images.githubusercontent.com/26584973/172003598-8cd48f64-2b68-4caf-b84f-56c6e7593b17.png)
</details>

**Comment by ishiharas** [comment]
Hello again, 
actually I'm using typescript `"typescript": "^4.7.2"` from the dev dependencies, but even after downgrading to `4.6.3 ` the methods didn't appear for me. My VS-Code version is `4.6.3` in case you wonder.
So the typescript version doesn't seem to be the issue here. 

Can you show me your package.json and tsconfig.json, so we can compare those? I guess, that's the only possible difference left?

<details><summary>tsconfig.json</summary>
<p>

```{
  "compilerOptions": {
    "module": "commonjs",
    "esModuleInterop": true,
    "target": "es6",
    "moduleResolution": "node",
    "sourceMap": true,
    "outDir": "dist"
  },
  "lib": ["es2015"]
}
```

</p>
</details>

<details><summary>package.json</summary>
<p>

```{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "src/main.ts",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "npx ts-node src/main.ts"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@types/cheerio": "^0.22.31",
    "@types/node": "^17.0.39",
    "ts-node": "^10.8.0",
    "typescript": "^4.7.2"
  },
  "dependencies": {
    "axios": "^0.27.2",
    "cheerio": "^1.0.0-rc.11",
    "xdccjs": "^4.4.19"
  }
}
```

</p>
</details>

Thank you for taking your time!

**Comment by JiPaix** [comment]
fixed by 6c9088147a78f5a8f17c64bbc3658eceb1f67647
[binaries](https://github.com/JiPaix/xdccJS/releases/tag/v4.4.20)  
[npm](https://www.npmjs.com/package/xdccjs/v/4.4.20)

the event types weren't exposed :man_facepalming:
